File: OC/ViewController.m
//
//  ViewController.m
//  OC
//
//  Created by lishuo on 16/3/17.
//  Copyright © 2016年 李硕. All rights reserved.
//

#import "ViewController.h"
#import "CycleScrollView.h"
@interface ViewController ()
@property (nonatomic, strong) CycleScrollView *cv;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.cv = [[CycleScrollView alloc] initWithFrame:CGRectMake(0, 20, 320, 240)];
    
    [self.view addSubview:self.cv];
    self.cv.backgroundColor = [UIColor grayColor];
    self.cv.showPageControl = true;
}

- (void)viewDidLayoutSubviews {
    [super viewDidLayoutSubviews];
    UIView *rview = [[UIView alloc] init];
    rview.backgroundColor = [UIColor redColor];
    
    UIView *gview = [[UIView alloc] init];
    gview.backgroundColor = [UIColor greenColor];
    
    UIView *bview = [[UIView alloc] init];
    bview.backgroundColor = [UIColor blueColor];
    
    UIView *yview = [[UIView alloc] init];
    yview.backgroundColor = [UIColor yellowColor];
    
    UIView *pview = [[UIView alloc] init];
    pview.backgroundColor = [UIColor purpleColor];
    
    self.cv.displaySource = @[rview,gview,bview,yview,pview];
    [self.cv setScrollTimeinterval:3];
}

@end

File: CycleScrollView/ViewController.swift
//
//  ViewController.swift
//  CycleScrollView
//
//  Created by 李硕 on 15/12/30.
//  Copyright © 2015年 李硕. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
       let csv = CycleScrollView(frame: CGRectMake(10, 300, 300, 100))
        let view1 = UIView()
        let view2 = UIView()
        let view3 = UIView()
        let view4 = UIView()

        view1.backgroundColor = UIColor.redColor()
        view2.backgroundColor = UIColor.blueColor()
        view3.backgroundColor = UIColor.yellowColor()
        view4.backgroundColor = UIColor.cyanColor()
        csv.displaySource = [view1,view2,view3,view4]
        
        csv.scrollTimeinterval = 5.0
        self.view.addSubview(csv)
        csv.showPageControl = true
        
    }
    
    @IBOutlet var csv: CycleScrollView!

    override func viewDidLayoutSubviews() {
        let view1 = UIView()
        let view2 = UIView()
        let view3 = UIView()
        let view4 = UIView()

        view1.backgroundColor = UIColor.redColor()
        view2.backgroundColor = UIColor.blueColor()
        view3.backgroundColor = UIColor.yellowColor()
        view4.backgroundColor = UIColor.cyanColor()
        csv.displaySource = [view1,view2,view3,view4]

        csv.scrollTimeinterval = 8.0
        csv.showPageControl = true
    }


    

   

}

